### DISCUSSION TITLE
Github action step getting stuck

### DISCUSSION THREAD

**Discussion by ErinRenNumber1** [community/community#27104] [Actions]
<p>Hi there:</p>
<p>I am very new to Github actions, and just trying to learn how to create the steps.  I am creating my very first Github action here, and one of my steps hangs.  I wonder if anyone can help me to understand what I am missing.  <a href="https://github.com/ErinRenNumber1/erin-ren-sandbox/runs/2729125375?check_suite_focus=true#step:5:5" class="inline-onebox" rel="noopener nofollow ugc">Create github-actions.yml · ErinRenNumber1/erin-ren-sandbox@8d0a9c0 · GitHub</a></p>
<p>Basically, I am trying to download shopify CLI, and try to print the version.  On my local machine, <code>shopify version</code> prints correctly.  I don’t understand why it hangs as a step here.  Any help would be appreciated!</p>

**Comment by jsoref** [comment, marked as answer]
<p>The problem is that <code>shopify *</code> wants to ask interactive questions at first launch even when there isn’t a tty (this is a bug).</p>
<pre><code class="lang-auto"> % shopify version
? Would you like to enable anonymous usage reporting?
If you select “Yes”, we’ll collect data about which commands you use and which errors you encounter.
Sharing this anonymous data helps Shopify improve this tool.
 (Choose with ↑ ↓ ⏎)
  1. yes
&gt; 2. no
</code></pre>
<p>Fwiw, I tried:</p>
<pre><code class="lang-auto">shopify config analytics off &lt; /dev/null
</code></pre>
<p>And that didn’t work either.</p>
<p>It is possible to drop a config file into place in order to bypass the (invisible) prompt:</p><aside class="onebox githubpullrequest">
  <header class="source">
      <a href="https://github.com/ErinRenNumber1/erin-ren-sandbox/pull/1" target="_blank" rel="noopener nofollow ugc">github.com/ErinRenNumber1/erin-ren-sandbox</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Pull Request">
    <svg width="60" height="60" class="github-icon" viewbox="0 0 12 16" aria-hidden="true"><path d="M11 11.28V5c-.03-.78-.34-1.47-.94-2.06C9.46 2.35 8.78 2.03 8 2H7V0L4 3l3 3V4h1c.27.02.48.11.69.31.21.2.3.42.31.69v6.28A1.993 1.993 0 0 0 10 15a1.993 1.993 0 0 0 1-3.72zm-1 2.92c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zM4 3c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v6.56A1.993 1.993 0 0 0 2 15a1.993 1.993 0 0 0 1-3.72V4.72c.59-.34 1-.98 1-1.72zm-.8 10c0 .66-.55 1.2-1.2 1.2-.65 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/ErinRenNumber1/erin-ren-sandbox/pull/1" target="_blank" rel="noopener nofollow ugc">suppress shopify config prompt -- we're unattended</a>
    </h4>

    <div class="branches">
      <code>ErinRenNumber1:master</code> ← <code>jsoref:unattended</code>
    </div>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2021-06-03" data-time="04:39:08" data-timezone="UTC">04:39AM - 03 Jun 21 UTC</span>
      </div>

      <div class="user">
        <a href="https://github.com/jsoref" target="_blank" rel="noopener nofollow ugc">
          <img alt="jsoref" src="https://user-images.githubusercontent.com/2119212/181101706-06141bdf-0c37-43b8-bd19-4478b01927ea.png" class="onebox-avatar-inline" width="20" height="20">
          jsoref
        </a>
      </div>

      <div class="lines" title="1 commits changed 1 files with 2 additions and 0 deletions">
        <a href="https://github.com/ErinRenNumber1/erin-ren-sandbox/pull/1/files" target="_blank" rel="noopener nofollow ugc">
          <span class="added">+2</span>
          <span class="removed">-0</span>
        </a>
      </div>
    </div>
  </div>
</div>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by jsoref** [comment]
<aside class="onebox githubissue">
  <header class="source">
      <a href="https://github.com/Shopify/shopify-app-cli/issues/1267" target="_blank" rel="noopener nofollow ugc">github.com/Shopify/shopify-app-cli</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/Shopify/shopify-app-cli/issues/1267" target="_blank" rel="noopener nofollow ugc">shopify-cli requires a terminal for first launch</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2021-06-03" data-time="04:50:27" data-timezone="UTC">04:50AM - 03 Jun 21 UTC</span>
      </div>


      <div class="user">
        <a href="https://github.com/jsoref" target="_blank" rel="noopener nofollow ugc">
          <img alt="jsoref" src="https://user-images.githubusercontent.com/2119212/181101709-35a473ad-9e84-417f-a4ec-f2b8786bd1f6.png" class="onebox-avatar-inline" width="20" height="20">
          jsoref
        </a>
      </div>
    </div>

    <div class="labels">
    </div>
  </div>
</div>

  <div class="github-row">
    <p class="github-body-container"># Issue summary

Running [shopify using GitHub Actions](https://github.communi<span class="show-more-container"><a href="" rel="noopener" class="show-more">…</a></span><span class="excerpt hidden">ty/t/github-action-step-getting-stuck/183184) hangs.

## Expected behavior

1. If there isn't a `tty` (e.g. in GitHub Actions), the program shouldn't stall waiting for one.
2. Running `shopify config analytics off` should work even if there isn't a tty (this *ALSO* stalls).

## Actual behavior

Program stalls without producing any output, leaving users scratching their heads.

## Workaround

After divining the path to the config file (`~/.config/shopify/config`), one can create the file and fill it with:
```ini
[analytics]
enabled = false
```
before running `shopify`. This works on both macOS and Linux.

## Steps to reproduce the problem

1. Locally clone https://github.com/ErinRenNumber1/erin-ren-sandbox
2. Create a GitHub repository
3. Push your local clone to your newly created GitHub repository (this should trigger the action)
4. You will see something like: https://github.com/ErinRenNumber1/erin-ren-sandbox/actions/runs/899843167

## Reduced test case

https://github.com/ErinRenNumber1/erin-ren-sandbox/actions/runs/899843167/workflow

## Specifications

- App type: Ruby Gem Cli (?)
- Operating System: macOS / Linux (reproduced using nektos/act w/ ubuntu-latest)
- Shell: bash
- Ruby version (ruby -v):
  - macOS: `ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin19]`
  - Ubuntu: `Unpacking ruby2.7 (2.7.0-5ubuntu1.4) ...`</span></p>
  </div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by ErinRenNumber1** [comment]
<p>Thank you so much <a class="mention" href="/jsoref">@jsoref</a>!  :smiley:</p>
